sprocs/variants_insert.sql

```diff
@@ -35,16 +38,17 @@ BEGIN
     IF instance_question_id IS NOT NULL THEN
         PERFORM instance_questions_lock(instance_question_id);
 
-        SELECT           q.id,    u.user_id,                  ai.id,                   ci.id
-        INTO real_question_id, real_user_id, assessment_instance_id, real_course_instance_id
+        SELECT           q.id,          gr.id,    u.user_id,                  ai.id,                   ci.id
+        INTO real_question_id, real_group_id, real_user_id, assessment_instance_id, real_course_instance_id
         FROM
             instance_questions AS iq
             JOIN assessment_questions AS aq ON (aq.id = iq.assessment_question_id)
             JOIN questions AS q ON (q.id = aq.question_id)
             JOIN assessment_instances AS ai ON (ai.id = iq.assessment_instance_id)
             JOIN assessments AS a ON (a.id = ai.assessment_id)
             JOIN course_instances AS ci ON (ci.id = a.course_instance_id)
-            JOIN users AS u ON (u.user_id = ai.user_id)
+            LEFT OUTER JOIN groups AS gr ON (gr.id = ai.group_id AND gr.deleted_at IS NULL)
+            LEFT OUTER JOIN users AS u ON (u.user_id = ai.user_id)
```

Comment:
We normally don't write OUTER in this situation in PL.

```suggestion
            LEFT JOIN groups AS gr ON (gr.id = ai.group_id AND gr.deleted_at IS NULL)
            LEFT JOIN users AS u ON (u.user_id = ai.user_id)
```